### PR TITLE
Add MooseDocs documentation generation to SALAMANDER contributing guidelines

### PR DESCRIPTION
**Pull request by cticenhour**
This also necessitated the addition of several items in the main webpage config.yml file, mostly from the python modules documentation, in order to give appropriate context to MooseDocs, etc.

Refs #2

### CODE REVIEW

**Comment by cticenhour** [comment]
@simopier this is passing and ready for review whenever you have time!

**Comment by cticenhour** [comment]
@gsgall Take a look at the new additions to the page as well. Do these changes functionally make sense to you (and a representative new SALAMANDER developer)?

**Review by gsgall — COMMENTED** [review]
Looks good to me.  I think this is a really helpful addition to the website

**Review by gsgall — APPROVED** [review]

File: doc/content/sqa/contributing.md [hunk 1]
```diff
@@ -165,7 +165,71 @@ unrelated changes all at once.
...
+Note that certain items, such as the source code description, object parameters, inputs in which the
+object is used, and child objects are filled in automatically using the [MooseDocs/extensions/appsyntax.md].
+The lines leading with `!!` as well as the `!alert` extension command should be removed and replaced
+with relevant documentation regarding how to use the object.
+
+!alert! note title=Use good documentation practices!
+When creating documentation, it is particularly important to note any design limitations or assumptions
+as well as best practices to apply when using the object.
+!alert-end!
+
```
Comment (simopier) [inline doc/content/sqa/contributing.md:226]:
```suggestion

Again, make sure to commit these changes using the instructions above before pushing your branch.
```

**Review by simopier — CHANGES_REQUESTED** [review]
Thank you Casey, this is awesome. I only have a minor suggestion.

File: doc/content/sqa/contributing.md [hunk 2]
```diff
@@ -165,7 +165,72 @@ unrelated changes all at once.
...
+Again, make sure to commit these changes using the instructions above before pushing your branch.
+## Pushing and submitting FENIX changes
```
Comment (cticenhour) [inline doc/content/sqa/contributing.md]:
```suggestion
Again, make sure to commit these changes using the instructions above before pushing your branch.

## Pushing and submitting FENIX changes
```

**Review by simopier — APPROVED** [review]
Thank you Casey!

**Pull request merged**